include/onnxruntime/core/graph/graph.h

```diff
@@ -342,52 +342,45 @@ class Node {
   /** Gets the number of output edges from this Node */
   size_t GetOutputEdgesCount() const noexcept { return relationships_.output_edges.size(); }
 
-  /** Add an attribute to this Node with specified attribute name and value. */
-  void AddAttribute(std::string attr_name, const ONNX_NAMESPACE::AttributeProto& value);
-  void AddAttribute(std::string attr_name, ONNX_NAMESPACE::AttributeProto&& value);
+  /** Adds an AttributeProto to this Node.
+  @remarks The attribute name is used as the key in the attribute map. */
+  void AddAttributeProto(ONNX_NAMESPACE::AttributeProto value);
 
```

Comment:
Do we always want to copy here? A graph attribute can potentially be quite large (not sure if that is a deep copy in the protobuf implementation though).

Reply:
It supports moving too. Usage should be similar to the previous two overloads with `const&` and `&&`.